Sources/SpeziOnboarding/OnboardingFlow/OnboardingNavigationPath.swift

```diff
@@ -97,30 +120,180 @@ public class OnboardingNavigationPath {
+    
+    
     /// An `OnboardingNavigationPath` represents the current navigation path within the ``OnboardingStack``.
     /// - Parameters:
     ///   - views: SwiftUI `View`s that are declared within the ``OnboardingStack``.
-    ///   - complete: An optional SwiftUI `Binding` that is injected by the ``OnboardingStack``.
+    ///   - isComplete: An optional SwiftUI `Binding` that is injected by the ``OnboardingStack``.
     ///     Is managed by the ``OnboardingNavigationPath`` to indicate whether the onboarding flow is complete.
-    ///   - startAtStep: An optional SwiftUI (Onboarding) `View` type indicating the first to-be-shown step of the onboarding flow.
-    init(views: [any View], complete: Binding<Bool>?, startAtStep: (any View.Type)?) {
-        self.complete = complete
-        updateViews(with: views)
-
+    ///   - startAtStep: Optionally, the step the OnboardingNavigationPath should initially move to.
+    func configure(elements: [_OnboardingFlowViewCollection.Element], isComplete: Binding<Bool>?, startAtStep: StepReference?) {
+        didConfigure = true
+        self.isComplete = isComplete
+        updateViews(with: elements)
         // If specified, navigate to the first to-be-shown onboarding step
-        if let startAtStep {
-            append(startAtStep)
+        switch startAtStep {
+        case nil:
+            break
+        case .viewType(let viewType):
+            appendStep(viewType)
+        case .identifier(let hashable):
+            appendStep(hashable)
         }
     }
+    
+    /// Internal function used to update the onboarding steps within the ``OnboardingNavigationPath`` if the
+    /// result builder associated with the ``OnboardingStack`` is reevaluated.
+    ///
+    /// This may be the case with `async` properties that are stored as a SwiftUI `State` in the respective view.
+    ///
+    /// - Parameters:
+    ///   - views: The updated `View`s from the ``OnboardingStack``.
+    func updateViews(with elements: [_OnboardingFlowViewCollection.Element]) {
+        do {
+            // Ensure that the incoming navigation stack elements are all unique.
+            // Note: we don't need to worry about collisions between OnboardingFlow-provided
+            // views and manually-added custom views added, since the non-custom ones will
+            // always also be identified by their source location, which is never the case for the custom ones.
+            var identifiersSeenSoFar = Set<OnboardingStepIdentifier>()
+            for element in elements {
+                let identifier = OnboardingStepIdentifier(element: element, isCustom: false)
+                guard identifiersSeenSoFar.insert(identifier).inserted else {
+                    let conflictingIdentifier = identifiersSeenSoFar.first(where: { $0 == identifier })
+                    preconditionFailure("""
+                        SpeziOnboarding: OnboardingStack contains elements with duplicate onboarding step identifiers.
```

Comment:
Great guard here, should enable devs to easily catch such issues in the dev process. We had issues in CS342 where students didn't understand the faulty behavior of the `OnboardingStack` when two Views of the same type where present (now also mitigated by the source location), such a guard helps a lot with that!